packages/flutter/lib/src/services/spell_check.dart

```diff
@@ -0,0 +1,213 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui';
+
+import 'package:flutter/foundation.dart';
+import 'system_channels.dart';
+
+/// A data structure representing a range of misspelled text and the suggested
+/// replacements for this range.
+///
+/// For example, one [SuggestionSpan] of the
+/// [List<SuggestionSpan>] suggestions of the [SpellCheckResults] corresponding
+/// to "Hello, wrold!" may be:
+/// ```dart
+/// SuggestionSpan(TextRange(7, 12), List<String>.from["word, world, old"])
+/// ```
+@immutable
+class SuggestionSpan {
+  /// Creates a span representing a misspelled range of text and the replacements
+  /// suggested by a spell checker.
+  ///
+  /// The [range] and replacement [suggestions] must all not
+  /// be null.
+  const SuggestionSpan(this.range, this.suggestions)
+      : assert(range != null),
+        assert(suggestions != null);
+
+  /// The misspelled range of text.
+  final TextRange range;
+
+  /// The alternate suggestions for the misspelled range of text.
+  final List<String> suggestions;
+
+  @override
+  bool operator ==(Object other) {
+    if (identical(this, other)) {
+        return true;
+    }
+
+    return other is SuggestionSpan &&
```

Comment:
Is it better to check the equality of the runtimeTypes instead of using `is`?  I've seen the runtimeType thing in several other `==` overrides ([example](https://github.com/flutter/flutter/blob/2a8a0890e25865b63deec46f4dd6e8c53650b73c/packages/flutter/lib/src/painting/box_decoration.dart#L324)).  Not sure if we prefer one or the other.

Reply:
I found the answer! https://dart.dev/guides/language/language-tour#getting-an-objects-type